matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/homeserver/DefaultHomeServerCapabilitiesService.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.matrix.android.internal.session.homeserver
+
+import com.zhuinden.monarchy.Monarchy
+import im.vector.matrix.android.api.session.homeserver.HomeServerCapabilities
+import im.vector.matrix.android.api.session.homeserver.HomeServerCapabilitiesService
+import im.vector.matrix.android.internal.database.mapper.HomeServerCapabilitiesMapper
+import im.vector.matrix.android.internal.database.model.HomeServerCapabilitiesEntity
+import im.vector.matrix.android.internal.database.query.getOrCreate
+import javax.inject.Inject
+
+internal class DefaultHomeServerCapabilitiesService @Inject constructor(private val monarchy: Monarchy) : HomeServerCapabilitiesService {
+
+    override fun getHomeServerCapabilities(): HomeServerCapabilities {
+        var entity: HomeServerCapabilitiesEntity? = null
+        monarchy.doWithRealm { realm ->
+            entity = HomeServerCapabilitiesEntity.getOrCreate(realm)
+        }
+
+        return entity?.let { HomeServerCapabilitiesMapper.map(it) } ?: HomeServerCapabilities(HomeServerCapabilities.MAX_UPLOAD_FILE_SIZE_UNKNOWN)
```

Comment:
I was using if/else at first, but the compiler was complaining that `entity` is changed by a capture closure and can be null even if tested not null... I wanted to avoid using `!!`. I've updated this code using `with`